packages/babel-helpers/src/helpers/extends.ts

```diff
@@ -0,0 +1,48 @@
+/* @minVersion 7.0.0-beta.0 */
+
+export default function _extends<T extends {}, U>(
+  this: any,
+  target: T,
+  source: U,
+): T & U;
+export default function _extends<T extends {}, U, V>(
+  this: any,
+  target: T,
+  source1: U,
+  source2: V,
+): T & U & V;
+export default function _extends<T extends {}, U, V, W>(
+  this: any,
+  target: T,
+  source1: U,
+  source2: V,
+  source3: W,
+): T & U & V & W;
+export default function _extends(
+  this: any,
+  target: object,
+  ...sources: any[]
+): any;
+export default function _extends(this: any) {
+  // @ts-expect-error explicitly assign to function
+  _extends = Object.assign
+    ? // need a bind because https://github.com/babel/babel/issues/14527
+      // @ts-expect-error -- intentionally omitting the argument
+      Object.assign.bind(/* undefined */)
```

Comment:
If `undefined` is explicitly given, the generated size will increase by about 6 bytes. So I omit the `undefined` and add a `ts-expect-error`.